Scripts/datatypes/purpose.py

```diff
@@ -113,15 +113,23 @@ def transform_impedance(self, impedance):
         mapping = self.zone_data.aggregations.municipality_centre_mapping
         day_imp = {}
         for mode in self.impedance_share:
+            outward_sum = 0
+            return_sum = 0
             day_imp[mode] = defaultdict(float)
             ass_class = mode.replace("pax", assignment_classes[self.name])
             for time_period in self.impedance_share[mode]:
                 for mtx_type in impedance[time_period]:
                     if ass_class in impedance[time_period][mtx_type]:
-                        share = self.impedance_share[mode][time_period]
                         imp = impedance[time_period][mtx_type][ass_class]
+                        share = self.impedance_share[mode][time_period]
+                        outward_sum += share[0]
+                        return_sum += share[1]
                         day_imp[mode][mtx_type] += share[0] * imp[rows, cols]
                         day_imp[mode][mtx_type] += share[1] * imp[cols, rows].T
+            if abs(outward_sum/len(impedance[time_period]) - 1) > 0.001:
+                raise ValueError(f"False outward impedance shares: {self.name} : {mode}")
+            if abs(return_sum/len(impedance[time_period]) - 1) > 0.001:
+                raise ValueError(f"False return impedance shares: {self.name} : {mode}")
```

Comment:
- We cannot divide the sum by `impedance[time_period]`, because the mode does not always have all impedance types. We could divide by `day_imp[mode]` instead.
- In Veeti's model, we only have outward impedance, that we want to sum to 2. So I think we could have just one comparison, checking that outward and return combined sums to 2.